xarray/core/netcdftimeindex.py

```diff
@@ -0,0 +1,180 @@
+import re
+from datetime import timedelta
+
+import numpy as np
+import pandas as pd
+
+from pandas.lib import isscalar
+
+
+def named(name, pattern):
+    return '(?P<' + name + '>' + pattern + ')'
+
+
+def optional(x):
+    return '(?:' + x + ')?'
+
+
+def trailing_optional(xs):
+    if not xs:
+        return ''
+    return xs[0] + optional(trailing_optional(xs[1:]))
+
+
+def build_pattern(date_sep='\-', datetime_sep='T', time_sep='\:'):
+    pieces = [(None, 'year', '\d{4}'),
```

Comment:
FYI NCAR's [TraCE](http://www.cgd.ucar.edu/ccr/TraCE/) simulation project is a 21k yr paleoclimate simulation.  Not sure how they handle calendars/times.  I know somebody who has analyzed data from this simulation; will ask what it looks like.